test.disabled/ejabberd_tests/tests/inbox_SUITE.erl

```diff
@@ -0,0 +1,601 @@
+  mark_last_muclight_system_message(User, ExpectedCount, <<"displayed">>).
+
+mark_last_muclight_system_message(User, ExpectedCount, MarkerType) ->
+  GetInbox = get_inbox_stanza(),
+  escalus:send(User, GetInbox),
+  Stanzas = escalus:wait_for_stanzas(User, binary_to_integer(ExpectedCount)),
+  ResIQ = escalus:wait_for_stanza(User),
+  ExpectedCount = get_inbox_count(ResIQ),
+  LastMsg = lists:last(Stanzas),
+  [InnerMsg] = get_inner_msg(LastMsg),
+  MsgId = exml_query:attr(InnerMsg, <<"id">>),
+  From = exml_query:attr(InnerMsg, <<"from">>),
+  ChatMarker =
+    set_type(escalus_stanza:chat_marker(From,MarkerType, MsgId), <<"groupchat">>),
+  escalus:send(User, ChatMarker).
+
+
+
+create_room_send_msg_check_inbox(Owner, MemberList, RoomName, Msg, Id) ->
+  RoomJid = room_bin_jid(RoomName),
+  OwnerJid = lbin(escalus_client:short_jid(Owner)),
+  create_room_and_check_inbox(Owner, MemberList, RoomName),
+  Stanza = escalus_stanza:set_id(
+    escalus_stanza:groupchat_to(RoomJid, Msg), Id),
+  escalus:send(Owner, Stanza),
+  foreach_recipient([Owner | MemberList],
+    fun(Stanza) ->
+    escalus:assert(is_groupchat_message, Stanza)
+    end),
+  %% send chat marker per each
+  OwnerRoomJid = <<RoomJid/binary,"/", OwnerJid/binary>>,
+  %% Owner sent the message so he has unread set to 0
+  check_inbox(Owner, <<"1">>, [{<<"0">>, OwnerRoomJid, OwnerJid, Msg}]),
+  foreach_check_inbox(MemberList, <<"1">>, <<"1">>, OwnerRoomJid, Msg).
+
+
+foreach_check_inbox(Users, Total, Unread, SenderJid, Msg) ->
+  [begin
+     UserJid = lbin(escalus_client:short_jid(U)),
+     check_inbox(U, Total, [{Unread, SenderJid,  UserJid, Msg}])
+    end || U <- Users].
+
+
+check_inbox(Client, ExpectedCount, MsgCheckList) when is_integer(ExpectedCount) ->
+  check_inbox(Client, integer_to_binary(ExpectedCount), MsgCheckList);
+check_inbox(Client, ExpectedCount, MsgCheckList) ->
+  check_inbox(Client, ExpectedCount, MsgCheckList, fun(X) -> X end).
+
+check_inbox(Client, ExpectedCount, MsgCheckList, AdditionalCheck) ->
+  GetInbox = get_inbox_stanza(),
+  escalus:send(Client, GetInbox),
+  Stanzas = escalus:wait_for_stanzas(Client, binary_to_integer(ExpectedCount)),
+  ResIQ = escalus:wait_for_stanza(Client),
+  ExpectedCount = get_inbox_count(ResIQ),
+  Merged = lists:zip(Stanzas, MsgCheckList),
+ [foreach_inbox_message(M, Unread, FromJid, ToJid, Content)
+    || {M, {Unread, FromJid, ToJid, Content}} <- Merged],
+  %% Apply additional asserts
+  [AdditionalCheck(M) || M <- Stanzas].
+
+
+foreach_inbox_message(Message, Unread, FromJid, ToJid, Content) ->
+  Unread = get_unread_count(Message),
+  true = escalus_pred:is_message(Message),
+  Unread = get_unread_count(Message),
+  [InnerMsg] = get_inner_msg(Message),
+  FromJid = exml_query:path(InnerMsg, [{attr, <<"from">>}]),
+  ToJid = exml_query:path(InnerMsg, [{attr, <<"to">>}]),
+  InnerContent = exml_query:path(InnerMsg, [{element, <<"body">>}, cdata], []),
+  InnerContent = Content.
```

Comment:
Badmatch here would include `Content` instead of `InnerContent`, so we won't know what inbox result has caused a failure.